core/arch/mangle_shared.c

```diff
@@ -464,14 +464,17 @@ insert_meta_call_vargs(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr,
         /* XXX PR 245936: let user decide whether to clean up?
          * i.e., support calling a stdcall routine?
          */
-#ifdef X86
+#if defined(X86)
         PRE(ilist, instr,
             INSTR_CREATE_lea(dcontext, opnd_create_reg(REG_XSP),
                              opnd_create_base_disp(REG_XSP, REG_NULL, 0,
                                                    stack_for_params, OPSZ_lea)));
-#elif defined(ARM)
-        /* FIXME i#1551: NYI on ARM */
-        ASSERT_NOT_IMPLEMENTED(false);
+#elif defined(AARCHXX)
+        PRE(ilist, instr, INSTR_CREATE_add(dcontext, opnd_create_reg(REG_XSP),
+                                           opnd_create_reg(REG_XSP),
+                                           OPND_CREATE_INT32(stack_for_params)));
```

Comment:
Please look at the docs for XINST_CREATE_add versus XINST_CREATE_add_s and the implementations.  We deliberately set it up that way after some discussion.